tests/_files/DataProviderRequiresPhpUnitTest.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\TestFixture;
+
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\RequiresPhpunit;
+use PHPUnit\Framework\TestCase;
+
+final class DataProviderRequiresPhpUnitTest extends TestCase
+{
+    public static function validProvider(): array
+    {
+        return [[true], [true]];
+    }
+
+    public function invalidProvider(): array
+    {
+        return [[true], [true]];
+    }
+
+    #[RequiresPhpunit('< 10')]
```

Comment:
I would like to see also another test with provider that always throws to verify it is really never called.

Reply:
Done!